Make ec2_vpc_nat_gateway results consistent

willthames opened the pull request:
##### SUMMARY
Ensure newly created NAT gateways get converted to snake dict
Remove custom code for generating snake dict and use camel_dict_to_snake_dict
Make use of required_if rather than bespoke parameter checks

##### ISSUE TYPE
 - Bugfix Pull Request

##### COMPONENT NAME
ec2_vpc_nat_gateway

##### ANSIBLE VERSION
```
ansible 2.4.0 (devel a5d6d3b9af) last updated 2017/04/27 09:28:12 (GMT +1000)
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/home/will/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /home/will/src/ansible/lib/ansible
  executable location = /home/will/src/ansible/bin/ansible
  python version = 2.7.13 (default, Jan 12 2017, 17:59:37) [GCC 6.3.1 20161221 (Red Hat 6.3.1-1)]

```


##### ADDITIONAL INFORMATION
Behaviour is already correct for existing nat gateways, which get converted to snake case, but new nat gateways don't get converted.

mattclay commented:
CI failure in unit tests:

```
    def test_convert_to_lower(self):
>       example = ng.DRY_RUN_GATEWAY_UNCONVERTED
E       AttributeError: 'module' object has no attribute 'DRY_RUN_GATEWAY_UNCONVERTED'

test/units/modules/cloud/amazon/test_ec2_vpc_nat_gateway.py:268: AttributeError
```

Commit:
Make ec2_vpc_nat_gateway results consistent

Ensure newly created NAT gateways get converted to snake dict
Remove custom code, and associated test, for generating snake
dict and use `camel_dict_to_snake_dict`
Make use of `required_if` rather than bespoke parameter checks
Remove ec2_vpc_nat_gateway from pep8 legacy files list

s-hertel commented:
shipit

Fixes #24129

ryansb commented:
Related change: https://github.com/ansible/ansible/pull/36058
